src/core/meshcop/dtls.cpp

```diff
@@ -260,6 +260,17 @@ otError Dtls::Setup(bool aClient)
 #endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 #endif // OPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE
 
+#if defined(MBEDTLS_SSL_SRV_C) && defined(MBEDTLS_SSL_COOKIE_C)
+    mbedtls_ssl_cookie_init(&mCookieCtx);
+    if (!aClient)
+    {
+        rval = mbedtls_ssl_cookie_setup(&mCookieCtx, mbedtls_ctr_drbg_random, Random::Crypto::MbedTlsContextGet());
```

Comment:
This needs to move after the call to `mbedtls_ssl_config_defaults()`, which includes the following:
https://github.com/openthread/openthread/blob/b339a8946bfa8aa6fbf5ed967471f18a85fbad87/third_party/mbedtls/repo/library/ssl_tls.c#L9093-L9096